examples/plot_kernel_approximation.py

```diff
@@ -68,118 +142,92 @@
 
 # fit and predict using linear and kernel svm:
+svm_times, svm_scores = {}, {}
+for kernel, clf in svms.items():
+    score, svm_performance = timing(fit_score, clf,
+                                    (data_train, targets_train),
+                                    (data_test, targets_test))
+    svm_scores[kernel] = score
+    svm_times[kernel] = svm_performance
 
-kernel_svm_time = time()
-kernel_svm.fit(data_train, targets_train)
-kernel_svm_score = kernel_svm.score(data_test, targets_test)
-kernel_svm_time = time() - kernel_svm_time
-
-linear_svm_time = time()
-linear_svm.fit(data_train, targets_train)
-linear_svm_score = linear_svm.score(data_test, targets_test)
-linear_svm_time = time() - linear_svm_time
 
+# create timing, accuracy data for approximate kernel models
 sample_sizes = 30 * np.arange(1, 10)
-fourier_scores = []
-nystroem_scores = []
-fourier_times = []
-nystroem_times = []
-
+scores, times = defaultdict(list), defaultdict(list)
 for D in sample_sizes:
-    fourier_approx_svm.set_params(feature_map__n_components=D)
-    nystroem_approx_svm.set_params(feature_map__n_components=D)
-    start = time()
-    nystroem_approx_svm.fit(data_train, targets_train)
-    nystroem_times.append(time() - start)
-
-    start = time()
-    fourier_approx_svm.fit(data_train, targets_train)
-    fourier_times.append(time() - start)
-
-    fourier_score = fourier_approx_svm.score(data_test, targets_test)
-    nystroem_score = nystroem_approx_svm.score(data_test, targets_test)
-    nystroem_scores.append(nystroem_score)
-    fourier_scores.append(fourier_score)
-
-# plot the results:
-plt.figure(figsize=(8, 8))
-accuracy = plt.subplot(211)
-# second y axis for timeings
-timescale = plt.subplot(212)
-
-accuracy.plot(sample_sizes, nystroem_scores, label="Nystroem approx. kernel")
-timescale.plot(sample_sizes, nystroem_times, '--',
-               label='Nystroem approx. kernel')
-
-accuracy.plot(sample_sizes, fourier_scores, label="Fourier approx. kernel")
-timescale.plot(sample_sizes, fourier_times, '--',
-               label='Fourier approx. kernel')
-
-# horizontal lines for exact rbf and linear kernels:
-accuracy.plot([sample_sizes[0], sample_sizes[-1]],
-              [linear_svm_score, linear_svm_score], label="linear svm")
-timescale.plot([sample_sizes[0], sample_sizes[-1]],
-               [linear_svm_time, linear_svm_time], '--', label='linear svm')
-
-accuracy.plot([sample_sizes[0], sample_sizes[-1]],
-              [kernel_svm_score, kernel_svm_score], label="rbf svm")
-timescale.plot([sample_sizes[0], sample_sizes[-1]],
-               [kernel_svm_time, kernel_svm_time], '--', label='rbf svm')
-
-# vertical line for dataset dimensionality = 64
-accuracy.plot([64, 64], [0.7, 1], label="n_features")
-
-# legends and labels
+    for feature_map, clf in approx_svm.items():
+        clf.set_params(feature_map__n_components=D)
+        fitted_clf, _time = timing(clf.fit, data_train, targets_train)
```

Comment:
In the original example, the linear and kernel svms are timed for both fit and score, but the approximate svms are timed for fitting only. I preserved that distinction for consistency, but I'm not sure it makes for a fair comparison in the final output. Timing only for fitting would also eliminate the need for the unappealing `fit_score` helper.